src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || a + b + c - Math.max(a, b, c) <= Math.max(a, b, c)
+    ) {
+      throw new Error('Wrong dimensions');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const { a, b, c } = this;
+    const semiPerimeter = (a + b + c) / 2;
+    const area = Math.sqrt(
+      semiPerimeter
+      * (semiPerimeter - a)
+      * (semiPerimeter - b)
+      * (semiPerimeter - c),
+    );
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    private radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid radius');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    private height: number,
+    private width: number,
+  ) {
+    if (Math.min(height, width) <= 0) {
+      throw new Error('Invalid rectangle');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(this.height * this.width * 100) / 100;
```

Comment:
seems like you have shared logic related to the rounding of area. consider moving it to helper.